### PR TITLE
doc: go dependency resolution

### PR DESCRIPTION
**Pull request by javierron**

### CODE REVIEW

**Comment by javierron** [comment]
@anderssonvivi Ready for you to make a pass on the text  :white_check_mark:

**Comment by monperrus** [comment]
Thanks @javierron  Is that fair to sau that the essence of the algorithm is to take the newest version of dependencies?
Is that by release date or by semver ordering?

File: dependency-resolution/index.md [hunk 1]
```diff
@@ -42,3 +42,18 @@ Thus, `Dx` is resolved to `v1.2.0`.
 
 ## Go
 
+To manage dependencies, Go uses _Modules_.
+In Go, a module is identified by its _module path_ and a _version_.
+A module used as a dependency is recorded in the `go.mod` file.
+For instance `github.com/BurntSushi/toml v1.3.2`, where `github.com/BurntSushi/toml` is the module path, and `v1.3.2` is the version (semver).
+
+To resolve the modules which are to be included in a build, it uses an algorithm called [Minimal Version Selection (MVS)](https://research.swtch.com/vgo-mvs) to generate a _build list_.
```
Comment (algomaster99) [inline dependency-resolution/index.md:50]:
```suggestion
To resolve the modules which are to be included in a build, it uses an algorithm called [Minimal Version Selection (MVS)](https://research.swtch.com/vgo-mvs) to generate a _build list_.
For dependency resolution, we only need to understand the first step in this algorithm i.e. - [Construct Build List](https://research.swtch.com/vgo-mvs#algorithm_1)
```

**Review by algomaster99 — COMMENTED** [review]
We could also generate the build list for the figure above and then show that `Dx v1.5.0` will be selected as it has the highest semver version.

I can write it once @anderssonvivi makes a pass.

**Comment by javierron** [comment]
@monperrus Yes, we can say that it takes the newest version reached by the dependency graph, by semver ordering.

But, another way to put it is that it takes the _oldest_ version that fits the requirements of all the included modules. This means it will not pick a newer module version which has been published, but not present in the dependency graph.

**Comment by algomaster99** [comment]
Thanks @javierron and @anderssonvivi !

**Pull request merged**